### PR TITLE
feat(transparency): add captum NoiseTunnel (SmoothGrad/VarGrad) support

### LINKED ISSUE #269: Support captum method-composition methods (NoiseTunnel/SmoothGrad; Neuron*, DataLoaderAttribution lower-priority)
Follow-up from #267 (which generalizes captum to `Layer*` methods). These three captum methods were left out of #267 because each needs a **new adapter contract element**, not a registry/normalization change — #267 stays inside the `method_class(model).attribute(inputs, target)` → input-shaped-tensor contract.

## Primary: NoiseTunnel (SmoothGrad / VarGrad) — real user demand

`NoiseTunnel.__init__(self, attribution_method: Attribution)` takes an **attribution instance, not a model**: `NoiseTunnel(IntegratedGradients(model))`. RAITAP constructs adapters as `method_class(model, **init_kwargs)`, so it can't build NoiseTunnel today (passing `model` where an `Attribution` is expected → type error).

Needs a **method-composition concept**: wrap a base attribution method (Saliency/IG/...) and apply SmoothGrad/VarGrad. This is the high-value item — SmoothGrad is a standard technique to denoise gradient attributions. It is stochastic (adds Gaussian noise), so once supported it auto-triggers the #251 reproducibility caveat by declaring `stochastic=True`.

Design questions:
- How to express "base method + noise wrapper" in transparency YAML (nested `_target_`? a `noise_tunnel:` block on an existing explainer?).
- `nt_type` (smoothgrad / smoothgrad_sq / vargrad), `nt_samples`, `stdevs` as `constructor`/`call` kwargs.

## Secondary (lower priority)

- **Neuron\*** (NeuronConductance, NeuronIntegratedGradients, NeuronGradient, ...): `attribute()` has a **required positional `neuron_selector`** (no default) and explains a chosen internal neuron, not the output class. Needs a new "which neuron" config concept + neuron-target output semantics. Niche (interpretability research).
- **DataLoaderAttribution**: ctor takes an `Attribution` (same model-vs-attribution mismatch as NoiseTunnel) **and** `attribute(dataloader, ...)` takes a `DataLoader`, not an input tensor — dataset-level output (GLOBAL-ish scope). Different input and output contract. Niche; overlaps RAITAP's deliberately-gated GLOBAL territory.

## Out of scope (separate, if ever)
GPU/tree-only SHAP and the dynamic unified `shap.Explainer` are tracked under #267's deferral, not here.

### PR DESCRIPTION
**Pull request by stanlrt**
## Summary

Adds captum `NoiseTunnel` (SmoothGrad / SmoothGrad-squared / VarGrad) support to `CaptumExplainer`, the Primary item of #269. NoiseTunnel wraps a base gradient attribution method and averages over noisy input copies to denoise it. It is configured entirely from transparency YAML.

Approach: route `CaptumExplainer` through the per-algorithm **invoker seam** (#266) exactly as `ShapExplainer` already does. NoiseTunnel's `NoiseTunnel(IG(model))` lifecycle (wrap an attribution instance, not a model) is the non-uniform construct-and-call the seam exists for.

- `refactor` — behavior-preserving extract of the Captum default path into `_default_captum_invoker`; `compute_attributions` becomes a thin dispatcher. Guarded by the existing captum suite (no behavior change for any current method).
- `feat` — `NoiseTunnel` registry entry (`GRADIENT`, `AUTOGRAD`, `stochastic=True`) with `_noise_tunnel_invoker`. `base_algorithm` is restricted to non-layer gradient methods via a registry-derived allow-list (today `Saliency`, `IntegratedGradients`), keeping the static entry honest and input-feature output-space inference correct. `stochastic=True` auto-fires the #251 reproducibility caveat (verified in a live run).
- `docs` / `chore` — NoiseTunnel section in `frameworks-and-libraries.md` and a self-contained `contributor-configs/noisetunnel-smoothgrad/` smoke config (bundled `imagenet_samples` + `vit_b_32`, no artifacts).

YAML:

```yaml
transparency:
  my_smoothgrad:
    _target_: CaptumExplainer
    algorithm: NoiseTunnel
    constructor:
      base_algorithm: IntegratedGradients
    call:
      target: 0
      nt_type: smoothgrad   # smoothgrad | smoothgrad_sq | vargrad
      nt_samples: 25
      stdevs: 0.1
```

`return_convergence_delta` through NoiseTunnel is not supported (no `extra_outputs`; a returned tuple fails safe via the arity-mismatch guard). Secondary `Neuron*` (now #293, P3) and `DataLoaderAttribution` stay deferred.

Closes #269.

## Checklist

- [x] **CI** — Required checks are green
- [x] **Breaking changes** — If this PR breaks compatibility (API, configs, file formats, etc.), the title uses `!` after the type or scope (e.g. `feat!:` or `feat(transparency)!:`). I understand that this change will have a **direct, disruptive impact** on package users. I ensured that **this change is absolutely necessary and cannot be delayed**.
- [x] **Contributor guide** — I’ve read [**Pull requests and commit messages**](https://caiivs.github.io/raitap/contributor/pull-requests.html) before requesting review.

## Optional

- [x] **Issue** _(optional)_ — A GitHub issues is linked to this PR ("Development" section in the right sidebar).
- [x] **Docs** _(optional)_ — User or contributor docs updated where needed.
- [x] **Tests** _(optional)_ — New or updated tests cover the change.

### CODE REVIEW

File: src/raitap/transparency/explainers/captum_explainer.py [hunk 1]
```diff
@@ -15,20 +15,80 @@
...
 
-from .base_explainer import AttributionOnlyExplainer
+from .base_explainer import AttributionInvokeCtx, AttributionOnlyExplainer
 
 if TYPE_CHECKING:
     import torch
     import torch.nn as nn
 
     from raitap.models.access import ExplanationModel
+    from raitap.transparency.contracts import InputSpec
```
[the remark this answers is not in the document]
Comment (stanlrt) [reply src/raitap/transparency/explainers/captum_explainer.py:25]: False positive. `InputSpec` is used at line 226 in `cast("InputSpec | None", input_spec)` — a string forward-ref annotation that pyright resolves (removing the import makes pyright fail with "InputSpec is not defined"). ruff and pyright are both clean on this file. This mirrors the sibling `ShapExplainer` (`cast("InputSpec | None", input_spec)`). Keeping the import; resolving.
